### PR TITLE
UI Notification Center

### PR DESCRIPTION
**Pull request by nhaagen**
Proposal for the NotificationCenter,  as described in http://www.ilias.de/docu/goto.php?target=wiki_1357_Page_Layout_Revision.

### CODE REVIEW

**Comment by klees** [comment]
Hi @Amstutz,

we started to build PRs for the Page Layout to have a place to explain the Notification Center and the Cockpit (as we have called it now).

Best regards!

File: src/UI/Component/Layout/Factory.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	 *   purpose: >
+	 *     A page-element describes a section or subsection of the ILIAS Application;
+	 *     the page thus is the total view upon ILIAS.
+	 *
+	 *
+	 * ----
+	 *
+	 * @return  \ILIAS\UI\Component\Layout\Page\Factory
+	 */
+	public function page();
```
Comment (Amstutz) [inline src/UI/Component/Layout/Factory.php:20]: See feedback of cockpit PR.

File: src/UI/Factory.php [hunk 2]
```diff
@@ -336,4 +336,15 @@ public function panel();
 	 **/
 	public function modal();
 
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: Layout elements
```
Comment (Amstutz) [inline src/UI/Factory.php:343]: See feedback of Cockpit PR.

File: src/UI/Component/Layout/Page/Factory.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The Notification Center concentrates the visualization of _any_
+     *     event-notification into one expandable glyph.
+     *     This unification removes the multitude of dedicated notification-icons
+     *     in favor of visual cleanliness as well as providing a designated
+     *     location for further extensions.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php:16]: I once heard the nice definition of "The notification center" offers possible "reactions" to notifications of the system for the user (e.g. new Mail) while the Cockpit offers action (e.g. open a course). Can we add that here somehow?

File: src/UI/Component/Layout/Page/Factory.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+     *      The Notification Center is visualized as a glyph with a counter.
+     *      Clicked, a pop-over list with the notifying services and their
+     *      respective counter-glyphs is expanded.
+     *
+     *   effect: >
+     *      All notifications, regardless of their origin, are summed up in
+     *      the counter of the Notification Center's glyph.
+     *      When clicked, a list is shown with all notifying services.
+     *      The entries each consist of the services' respective glyph, counter
+     *      and title.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php:28]: Pleace move this to the respective entry of this list if one will exist. If not, move to composition.

File: src/UI/Component/Layout/Page/Factory.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+     *      Entries as well can be clicked; the user is then directed to the
+     *      view of the service.
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php:30]: Or change the popovers content to some content offering direct interaction (e.g. see background tasks).

File: src/UI/Component/Layout/Page/Factory.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+     *        As the Awareness Tool does not actually target potentially
+     *        permanent notifications to the user and offers further functionality
+     *        when an entry is expanded, it cannot be properly enclosed in the
+     *        Notification Center and should reside next to it.
+     *
+     * rules:
+     *   usage:
+     *     1: There MUST be but one notification center on the page.
+     *   interaction:
+     *   composition:
```
Comment (Amstutz) [inline src/UI/Component/Layout/Page/Factory.php:43]: remove if empty.

Comment (Amstutz) [reply src/UI/Component/Layout/Page/Factory.php:43]: Add rules about accessibility here.

**Review by Amstutz — CHANGES_REQUESTED** [review]
This PR seems incomplete. Please add the complete interface. Also, add the mockups for a quick overview for other developers.

Further how will entries in the list of notifications be devised? Do they get their own entries, similar to the "Cockpit entries" or are they just generic items? What about the content of the popover, if one such entry is clicked upon? Please look at the mockups provided in the FR.

**Comment by klees** [comment]
We will open new PRs once our components are ready. Thx for Feedback!

**Pull request closed without merging**